This 'md' file is an overview of how to perform non metric multidimensional scaling (NMDS) using the vegan package with microbiome data originating from a phyloseq object.
I've also included a more thorough background description on  the 'sample_data' (i.e. meta data component) for this initial phyloseq object/2022 sample data. 
If you are interested in more background on how NMDS works, its strengths and weaknesses compared to other ordination methods,
see the following review: https://onlinelibrary.wiley.com/doi/full/10.1111/mec.13536

NMDS is an exploratory analysis. Strong for getting an overall sense of how different two (or more) microbial communities are based on some set of phylogenetic (i.e. rRNA genes) or
functional (metagenomic) datasets. 

First, load the following packages. 
```
library(vegan)
library(phyloseq)
library(tidyverse)
```
Load the workspace with the phyloseq object and other data (JRW_simplePhyloSeq_07102023.rds)
```
load("JRW_simplePhyloSeq_07102023.rds")
```
SINCE THIS IS BEING ASKED SEVERAL TIMES A DAY WE MIGHT AS WELL GET AHEAD OF OURSELVES AND REMOVE EUKARYOTIC SIGNAL.

```
ps.clean <- ps %>%
  taxa_prune("Chloroplast", "Order") %>%
  taxa_prune("Mitochondria", "Family")
```
We should probably rarefy the data (i.e. subsample to equal sequencing depths, eliminating samples beneath a certain threshold).
I'm choosing 1000 as the threashold here.

```
ps.clean.rare = rarefy_even_depth(ps.clean, 1000)
```
So this phyloseq object should now have 70 samples and 24842 OTUs. Note the changes from the original phyloseq object.
We've removed 5 samples that did not have at least 1000 sequences. 46149 ASVs were removed as well. This is a lot...
Most of the sequences removed will likely be lower abundance OTUs (less than 0.01%). These might be quite interesting for further analyses.
Just keep in mind that in these initial NMDS will be evaluating community structure differences based on more dominant taxa.

Recall, for this initial data set, we have three sampling times. I've made a column ('Dates') that indicates the sampling trip ('JRW1' is April 2022, 'JRW2' is June 2022, 'JRW3' is June 2022).
This column ('Dates') can accessed using the 'sample_data' function in phyloseq (we'll be doing this a lot). 
DO NOTE! Most the sample_data in 'ps' (and all objects extending from it, ps.clean) was  produced manually, by Tony, and is categorical in nature.
For example, there is a column called 'Environment' where i have assigned each sample to either 'Stream' or 'Well' based on whether the sample originated from
the stream channel or a well, respectively. Other examples include, 'Site' (RailRoad, Porter, LowerL), 'Time' (not a real GMT time, but rather which time during the two day diurnal sampling
we collected samples from the stream channel, DAWN (0500), MIDDAY (~1100/1200), DUSK (~1700), NIGHT (~2200)) , 'Color' (i assigned a color based on 'Stream' or 'Well', i think black and blue), 'Period_Color' (colors assigned based on time periods 'DAWN', 'MIDDAY', 'DUSK', and 'NIGHT', i think its orange, black, blue, red, respectively). 
