This 'md' file is an overview of how to perform non metric multidimensional scaling (NMDS) using the vegan package with microbiome data originating from a phyloseq object.
I've also included a more thorough background description on  the 'sample_data' (i.e. meta data component) for this initial phyloseq object/2022 sample data. 
If you are interested in more background on how NMDS works, its strengths and weaknesses compared to other ordination methods,
see the following review: https://onlinelibrary.wiley.com/doi/full/10.1111/mec.13536

NMDS is an exploratory analysis. Strong for getting an overall sense of how different two (or more) microbial communities are based on some set of phylogenetic (i.e. rRNA genes) or
functional (metagenomic) datasets. 

First, load the following packages. 
```
library(vegan)
library(phyloseq)
library(tidyverse)
```
Load the workspace with the phyloseq object and other data (JRW_simplePhyloSeq_07102023.rds)
```
load("JRW_simplePhyloSeq_07102023.rds")
```
SINCE THIS IS BEING ASKED SEVERAL TIMES A DAY WE MIGHT AS WELL GET AHEAD OF OURSELVES AND REMOVE EUKARYOTIC SIGNAL.

```
ps.clean <- ps %>%
  taxa_prune("Chloroplast", "Order") %>%
  taxa_prune("Mitochondria", "Family")
```
We should probably rarefy the data (i.e. subsample to equal sequencing depths, eliminating samples beneath a certain threshold).
I'm choosing 1000 as the threashold here.

```
ps.clean.rare = rarefy_even_depth(ps.clean, 1000)
```
So this phyloseq object should now have 70 samples and 24842 OTUs. Note the changes from the original phyloseq object.
We've removed 5 samples that did not have at least 1000 sequences. 46149 ASVs were removed as well. This is a lot...
Most of the sequences removed will likely be lower abundance OTUs (less than 0.01%). These might be quite interesting for further analyses.
Just keep in mind that in these initial NMDS will be evaluating community structure differences based on more dominant taxa.

Recall, for this initial data set, we have three sampling times. I've made a column ('Dates') that indicates the sampling trip ('JRW1' is April 2022, 'JRW2' is June 2022, 'JRW3' is June 2022).
This column ('Dates') can accessed using the 'sample_data' function in phyloseq (we'll be doing this a lot). 
DO NOTE! Most the sample_data in 'ps' (and all objects extending from it, ps.clean, ps.clean.rare) was  produced manually, by Tony, and is categorical in nature.
For example, there is a column called 'Environment' where i have assigned each sample to either 'Stream' or 'Well' based on whether the sample originated from
the stream channel or a well, respectively. Other examples include, 'Site' (RailRoad, Porter, LowerL), 'Time' (not a real GMT time, but rather which time during the two day diurnal sampling
we collected samples from the stream channel, DAWN (0500), MIDDAY (~1100/1200), DUSK (~1700), NIGHT (~2200)) , 'Color' (i assigned a color based on 'Stream' or 'Well', i think black and blue), 'Period_Color' (colors assigned based on time periods 'DAWN', 'MIDDAY', 'DUSK', and 'NIGHT', i think its orange, black, blue, red, respectively). 

In addition to categorical data, I added (again manually) continuous data (mainy chemical data collected from the chemical lab on the 8th floor). It should be fairly obvious which data these are, but essentially columns 12 to 28 (temperature, specific conductivity (SC), dissolved oxygen (DO), and other nitrogen information) is contained here. 

If anything here does not make sense just ask- again, this was object ('ps') was put together manually. 

OK- now we can start doing some NMDS analyses. Sort of. Actually, prior to doing an NMDS (or any other ordination) we have to produce distance matrix. There are several ways to do this.
Lets first start by looking at RailRoad data from the first sampling trip, and ask ourselves this: are there compositional differences between the stream channel samples and the well samples?
To do this, we first need to subset the 'ps.clean.rare', getting just time1 data from the railroad site.

```
#First get all the railroad samples
RR = subset_samples(ps.clean.rare, Site == "RailRoad")
#Now, get just time one samples.
RR_Time1 = subset_samples(RR, Date == "JRW1")
```
Another more direct way to subset:
```
RR_Time1 <- subset_samples(ps.clean.rare) %>%
 subset_samples(Date == "JRW1") %>%
 subset_samples(Site == "RailRoad")
```
NMDS functions, or any other ordination type, require a distance matrix as input. There are several different distance matrix types (bray-curtis, euclidean, jacarrd, ect.) we can produce.
For the sake of brevity, i'm not going to try and explain the strengths and weaknesses of each. For thorough discussion on these topics see:

https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2121141/
https://academic.oup.com/femsec/article/90/3/543/536864

Another distance metric that has become quite popular uses phylogenetic information (associated with our OTU table) to establish community structural differences. This is called unifrac.
https://journals.asm.org/doi/full/10.1128/AEM.71.12.8228-8235.2005

First we'll just use a non-phylogenetic, more 'classic', ordination approach using the 'vegdist' function in the vegan package.
'vegdist' uses the Bray-Curtis metric for establishing distances by default. 

```
RR_Time1.distance = vegdist(otu_table(RR_Time1))
```
Look at the distance in the R console (just type 'RR_Time1.distance'). You should see a triangular matrix. 
Note, phyloseq 'keeps' OTU tables in a transposed state, so we don't need to transpose the 'otu_table' prior to running the measurement.
More simply put, the data are already correctly positioned to look at sample to sample differences in composition. 

We can now use this distance matrix to produce an ordination. 
I prefer the 'monoMDS' function in vegan.

```
RR_Time1.distance.nmds = monoMDS(RR_Time1.distance)
```
This should run without a problem. type 'RR_Time1.distance.nmds' and look at the result:

```
RR_Time1.distance

Call:
monoMDS(dist = RR_Time1.distance) 

Non-metric Multidimensional Scaling

12 points, dissimilarity ‘bray’, call ‘vegdist(x = otu_table(RR_Time1))’

Dimensions: 2 
Stress:     0.07749763 
Stress type 1, weak ties
Scores scaled to unit root mean square, rotated to principal components
Stopped after 136 iterations: Stress nearly unchanged (ratio > sratmax)
```
The 'stress' of the result is low. Which is good. The lower the stress, the easier with which the ordination can be fit into an appropriate space.
More simply put, the lower stress the more confidence we have that the solution is appropriate. 

We can now plot the result.

```
plot(RR_Time1.distance.nmds)
```
This 'should' produce figure on the graph console in R studio. 
Another way to produce this figure would be do make a pdf and open it on the desktop.

```
pdf("/Users/stewartlab/Desktop/RR_Time1.distance.nmds.pdf")
plot(RR_Time1.distance.nmds)
dev.off()
```
