This 'md' file is an overview of how to perform non metric multidimensional scaling (NMDS) using the vegan package with microbiome data originating from a phyloseq object.
If you are interested in more background on how NMDS works, its strengths and weaknesses compared to other ordination methods,
see the following review: https://onlinelibrary.wiley.com/doi/full/10.1111/mec.13536

NMDS is an exploratory analysis. Strong for getting an overall sense of how different two (or more) microbial communities are based on some set of phylogenetic (i.e. rRNA genes) or
functional (metagenomic) datasets. 

First, load the following packages. 
```
library(vegan)
library(phyloseq)
library(tidyverse)
```
Load the workspace with the phyloseq object and other data (JRW_simplePhyloSeq_07102023.rds)
```
load("JRW_simplePhyloSeq_07102023.rds")
```
SINCE THIS IS BEING ASKED SEVERAL TIMES A DAY WE MIGHT AS WELL GET AHEAD OF OURSELVES AND REMOVE EUKARYOTIC SIGNAL.

```
ps.clean <- ps %>%
  taxa_prune("Chloroplast", "Order") %>%
  taxa_prune("Mitochondria", "Family")
```
We should probably rarefy the data (i.e. subsample to equal sequencing depths, eliminating samples beneath a certain threshold).
I'm choosing 1000 as the threashold here.

```
ps.clean.rare = rarefy_even_depth(ps.clean, 1000)
```
